stub/src/main/java/io/grpc/stub/AbstractAsyncStub.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2019 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.stub;
+
+import io.grpc.CallOptions;
+import io.grpc.Channel;
+import io.grpc.stub.ClientCalls.StubType;
+import javax.annotation.CheckReturnValue;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * Stub implementations for async stubs. Stub configuration is immutable; changing the
+ * configuration returns a new stub with updated configuration. Changing the configuration is cheap
+ * and may be done before every RPC, such as would be common when using {@link #withDeadlineAfter}.
+ *
+ * <p>Configuration is stored in {@link CallOptions} and is passed to the {@link Channel} when
+ * performing an RPC.
+ *
+ * <p>DO NOT MOCK: Customizing options doesn't work properly in mocks. Use InProcessChannelBuilder
+ * to create a real channel suitable for testing. It is also possible to mock Channel instead.
+ *
+ * @since 1.25.0
+ */
+@ThreadSafe
+@CheckReturnValue
+public abstract class AbstractAsyncStub<S extends AbstractAsyncStub<S>> extends AbstractStub<S> {
+
+  protected AbstractAsyncStub(Channel channel) {
+    this(channel, CallOptions.DEFAULT);
+  }
+
+  protected AbstractAsyncStub(Channel channel, CallOptions callOptions) {
+      super(channel, callOptions);
+  }
+
+  // TODO(jihuncho) javadoc
+  public static <T extends AbstractStub<T>> T newStub(StubFactory<T> factory, Channel channel) {
```

Comment:
s/AbstractStub/AbstractAsyncStub/

Reply:
removed unused newStub functions from AbstractStub to make it work.

Reply:
That doesn't sound right. We should get to the bottom of that.

Reply:
because they have same method signature after the type erasure, static "override". either we keep `newStub` methods in `AbstractStub` or we use `<T extends AbstractStub<T>>` in extended Stub classes.